File: QTD2.Domain/Entities/Core/EnablingObjective_Topic.cs
using System;
using System.Collections.Generic;

namespace QTD2.Domain.Entities.Core
{
    public class EnablingObjective_Topic : Common.Entity
    {
        public string Title { get; set; }

        public string Description { get; set; }

        public int SubCategoryId { get; set; }

        public int? Number { get; set; }

        public DateTime EffectiveDate { get; set; }

        public virtual EnablingObjective_SubCategory EnablingObjectives_SubCategory { get; set; }

        public virtual ICollection<EnablingObjective> EnablingObjectives { get; set; } = new List<EnablingObjective>();

        public virtual ICollection<CustomEnablingObjective> CustomEnablingObjectives { get; set; } = new List<CustomEnablingObjective>();

        public virtual ICollection<EnablingObjective_TopicHistory> EnablingObjective_TopicHistories { get; set; } = new List<EnablingObjective_TopicHistory>();
        public EnablingObjective_Topic(string description, int subCategoryId, int number, string title, DateTime effectiveDate)
        {
            Description = description;
            SubCategoryId = subCategoryId;
            Number = number;
            Title = title;
            EffectiveDate = effectiveDate;
        }

        public EnablingObjective_Topic()
        {
        }

        public EnablingObjective AddEnablingObective(EnablingObjective enablingObjective)
        {
            AddEntityToNavigationProperty<EnablingObjective>(enablingObjective);
            return enablingObjective;
        }
    }
}

File: QTD.SPA/src/app/_DtoModels/EnablingObjective_Topic/EnablingObjective_Topic.ts
import { EnablingObjective } from '../EnablingObjective/EnablingObjective';
import { EnablingObjective_SubCategory } from '../EnablingObjective_SubCategory/EnablingObjective_SubCategory';
import { Entity } from '../Entity';

export class EnablingObjective_Topic extends Entity{
  description!: string;
  subCategoryId!: any;
  number!: number;
  title!: string;
  effectiveDate!:any;
  enablingObjectives_SubCategory!: EnablingObjective_SubCategory;
  enablingObjectives!: EnablingObjective[];
}
